src/include/storage/projected_columns.h

```diff
@@ -13,17 +13,17 @@ namespace terrier::storage {
  * tuple slot, so it is more compact in layout than MaterializedColumns, which has to also store the
  * TupleSlot information for each tuple. The inner class RowView provides access to the underlying logical
  * projected rows with the same interface as a real ProjecetedRow.
- * -----------------------------------------------------------------------
- * | size | max_tuples | num_tuples | num_cols | col_id1 | col_id2 | ... |
- * -----------------------------------------------------------------------
- * | val1_offset | val2_offset | ... | TupleSlot_1 | TupleSlot_2 |  ...  |
- * -----------------------------------------------------------------------
- * | null-bitmap, col_id1 | val1, col_id1 | val2, col_id1 |      ...     |
- * -----------------------------------------------------------------------
- * | null-bitmap, col_id1 | val1, col_id2 | val2, col_id2 |      ...     |
- * -----------------------------------------------------------------------
- * |                                ...                                  |
- * -----------------------------------------------------------------------
+ * -------------------------------------------------------------------------------------
+ * | size | max_tuples | num_tuples | num_cols | attr_end[4] | col_id1 | col_id2 | ... |
```

Comment:
is attr_end supposed to be the array of attribute size? Why isn't this documented as attr_sizes?

Reply:
Nvm I get it, it's storing where the boundaries are.